modules/frontend/tracebyidsharding_test.go

```diff
@@ -38,22 +40,53 @@ func TestCreateBlockBoundaries(t *testing.T) {
 			name:        "multiple shards",
 			queryShards: 4,
 			expected: [][]byte{
-				{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},  // 0
-				{0x3f, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, // 0x3f = 255/4 * 1
-				{0x7e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, // 0x7e = 255/4 * 2
-				{0xbd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, // 0xbd = 255/4 * 3
+				{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+				{0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+				{0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+				{0xc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
 				{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
 			},
 		},
+		{
+			name:        "large number of evenly divisible shards",
+			queryShards: 255,
+		},
+		{
+			name:        "large number of not evenly divisible shards",
+			queryShards: 1111,
```

Comment:
Could this be tested with hardcoded expected values and a smaller non even value like 3 shards?  Instead of the code down below.

Reply:
Just curious, how is all this tested apart from the unit tests?

Reply:
The various shard sizes are not directly tested anywhere else that I'm aware of, but we do test the entire e2e stack here:

https://github.com/grafana/tempo/tree/main/integration/e2e

Reply:
So now that number of shards has changed, would it require changes in e2e tests too?

Reply:
The number of shards hasn't changed, but the maximum configurable number of shards has. We choose not to test every combination of parameters in the e2e tests b/c it would make for an exceptionally long test. The e2e tests are "smoke tests" to make sure that the basic functionality of Tempo is working.